### DISCUSSION TITLE
why Tensor type warning appears in google colab?

### DISCUSSION THREAD

**Discussion by falconlee236** [ai4co/rl4co#214] [Q&A]
I try to implement `1-quickstart.ipynb` in your repository to colab.  
but they exists type warning about that
```
Argument of type "Tensor" cannot be assigned to parameter "td" of type "TensorDict" in function "render"
  "Tensor" is incompatible with "TensorDict"(reportGeneralTypeIssues)
(variable) td_init: TensorDict
```

how can i fix that warning? 
<img width="846" alt="image" src="https://github.com/user-attachments/assets/a0c82651-57dd-481e-b4dd-97261d7237e2">

**Comment by fedebotu** [comment]
Hi @falconlee236 !

Does the problem appear in the cell above? (i.e. [here](https://rl4.co/examples/1-quickstart/#test-greedy-rollout-with-untrained-model-and-plot)? )


I cannot get any warning on my side

**Reply by falconlee236** [reply]
Are you use Colab with T4 GPU?
the error message shows that
```
Argument of type "Tensor" cannot be assigned to parameter "td" of type "TensorDict" in function "render"
  "Tensor" is incompatible with "TensorDict"(reportGeneralTypeIssues)
TensorDict: td_init

TensorDict with shape torch.Size([3])

No quick fixes available
```
![image](https://github.com/user-attachments/assets/a9f60b03-b81f-407f-8c1f-cbfd0542856f)

my code is this
```
# Greedy rollouts over untrained policy
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
td_init = env.reset(batch_size=[3]).to(device)
policy = policy.to(device)
out = policy(
    td_init.clone(),
    phase="test",
    decode_type="greedy",
    return_actions=True
)
actions_untrained = out["actions"].cpu().detach()
rewards_untrained = out["reward"].cpu().detach()

for i in range(3):
    print(f"Proble {i + 1} | Cost: {-rewards_untrained[i]:.3f}")
    env.render(td_init[i], actions_untrained[i])
```
@fedebotu

**Reply by fedebotu** [reply]
Yes, I used T4. Are you using the quick start notebook as is / did you modify anything in it? (it seems the same to me 🤔)

**Reply by falconlee236** [reply]
my notebook code is [1-quickstart.zip](https://github.com/user-attachments/files/16994181/1-quickstart.zip)

i am frustrated that i couldn't find why this error occurs

**Reply by fedebotu** [reply]
I uploaded your notebook to Google Drive and opened it with Colab. Then, I connected a runtime to T4 and ran all the notebook including rendering and training. It seems that I can't reproduce your error :thinking: 

The only thing that might be different could be the runtime restart: you need to install `rl4co` and then restart the runtime. Did you do it before running the other cells?

![image](https://github.com/user-attachments/assets/3ac4e873-3192-4756-ae07-74a12b52a694)

**Reply by falconlee236** [reply]
in your example code. In colab env, there must restart after installing rl4co library.

so i install rl4co first. then restart. finally run cell below that restart cell.
